lib/jxl/decode.cc

```diff
@@ -2092,7 +2092,7 @@ static JxlDecoderStatus HandleBoxes(JxlDecoder* dec) {
           dec->AdvanceInput(avail_codestream);
         }
 
-        if (dec->file_pos == dec->box_contents_end) {
+        if (dec->file_pos == dec->box_contents_end && dec->box_size != 0) {
```

Comment:
To be consistent with the rest of the file, please use !dec->box_contents_unbounded instead of dec->box_size != 0

Reply:
Done